leaderelection/context.go

```diff
@@ -53,10 +56,10 @@ func WithStandardLeaderElectorBuilder(ctx context.Context, kc kubernetes.Interfa
 	})
 }
 
-// WithStatefulSetLeaderElectorBuilder infuses a context with the ability to build
+// withStatefulSetElectorBuilder infuses a context with the ability to build
 // Electors which are assigned leadership based on the StatefulSet ordinal from
 // the provided component configuration.
-func WithStatefulSetLeaderElectorBuilder(ctx context.Context, cc ComponentConfig, ssc StatefulSetConfig) context.Context {
+func withStatefulSetElectorBuilder(ctx context.Context, cc ComponentConfig, ssc statefulSetConfig) context.Context {
```

Comment:
I think we might still want to make this public as autoscaler binary is not using the sharedmain. And only StatefulSet mode works for autoscaler. Falling back to Standard mode via `WithDynamicLeaderElectorBuilder` doesn't work in autoscaler case.

Reply:
I talked with Matt about it a bit. We can just call WithDynaminLeaderElectorBuilder and make it fail if SS builder is expected.